pkg/cloudprovider/aws/block_store.go

```diff
@@ -81,7 +81,30 @@ func (b *BlockStore) Init(config map[string]string) error {
 	return nil
 }
 
+func (b *BlockStore) isSnapshotEncrypted(snapshotID string) (encrypted bool, err error) {
+	req := &ec2.DescribeSnapshotsInput{
+		SnapshotIds: []*string{&snapshotID},
+	}
+
+	res, err := b.ec2.DescribeSnapshots(req)
+	if err != nil {
+		return false, errors.WithStack(err)
+	}
+
+	if len(res.Snapshots) != 1 {
+		return false, errors.Errorf("Expected one snapshot from DescribeSnaphots for snapshot ID %v, got %v", snapshotID, len(res.Snapshots))
+	}
+
+	snapshot := res.Snapshots[0]
+	return *snapshot.Encrypted, nil
+}
+
 func (b *BlockStore) CreateVolumeFromSnapshot(snapshotID, volumeType, volumeAZ string, iops *int64) (volumeID string, err error) {
+	encrypted, err := b.isSnapshotEncrypted(snapshotID)
+	if err != nil {
+		return "", errors.WithStack(err)
```

Comment:
you can just `return "", err` here since the error coming from `isSnapshotEncrypted` will already have a stack